src/reducerMaps/restaurants.js

```diff
@@ -124,12 +130,15 @@ export default new Map([
     })
   ],
   [ActionTypes.POST_VOTE, isFetching],
-  [ActionTypes.VOTE_POSTED, (state, action) =>
-    update(state, {
+  [ActionTypes.VOTE_POSTED, (state, action) => {
+    const updates = {
       isFetching: {
         $set: false
-      },
-      items: {
+      }
+    };
+
+    if (!(action.vote.id in state.items.entities.votes)) {
```

Comment:
It's possible in rare cases for `state.items.entities.votes` to not exist (see line 153). Make sure you check here as well. 

Also, don't use `in`. Check if `indexOf` is `-1` instead.